src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs

```diff
@@ -42,118 +42,82 @@ public void Abort()
-            bool disposeHandler = true;
+
+            bool tryDowngrade = false;
             try
             {
-                var request = new HttpRequestMessage(HttpMethod.Get, uri);
-                if (options._requestHeaders?.Count > 0) // use field to avoid lazily initializing the collection
-                {
-                    foreach (string key in options.RequestHeaders)
-                    {
-                        request.Headers.TryAddWithoutValidation(key, options.RequestHeaders[key]);
-                    }
-                }
 
-                // Create the security key and expected response, then build all of the request headers
-                KeyValuePair<string, string> secKeyAndSecWebSocketAccept = CreateSecKeyAndSecWebSocketAccept();
-                AddWebSocketHeaders(request, secKeyAndSecWebSocketAccept.Key, options);
-
-                // Create the handler for this request and populate it with all of the options.
-                // Try to use a shared handler rather than creating a new one just for this request, if
-                // the options are compatible.
-                if (options.Credentials == null &&
-                    !options.UseDefaultCredentials &&
-                    options.Proxy == null &&
-                    options.Cookies == null &&
-                    options.RemoteCertificateValidationCallback == null &&
-                    options._clientCertificates?.Count == 0)
+                while (true)
                 {
-                    disposeHandler = false;
-                    handler = s_defaultHandler;
-                    if (handler == null)
+                    try
                     {
-                        handler = new SocketsHttpHandler()
+                        HttpRequestMessage request;
+                        if (!tryDowngrade && options.HttpVersion == HttpVersion.Version20
+                            || (options.HttpVersion == HttpVersion.Version11 && options.HttpVersionPolicy == HttpVersionPolicy.RequestVersionOrHigher))
                         {
-                            PooledConnectionLifetime = TimeSpan.Zero,
-                            UseProxy = false,
-                            UseCookies = false,
-                        };
-                        if (Interlocked.CompareExchange(ref s_defaultHandler, handler, null) != null)
+                            request = new HttpRequestMessage(HttpMethod.Connect, uri);
+                            request.Version = HttpVersion.Version20;
+                            tryDowngrade = true;
+                        }
+                        else if (tryDowngrade || options.HttpVersion == HttpVersion.Version11)
                         {
-                            handler.Dispose();
-                            handler = s_defaultHandler;
+                            request = new HttpRequestMessage(HttpMethod.Get, uri);
+                            request.Version = HttpVersion.Version11;
```

Comment:
```suggestion
                            request = new HttpRequestMessage(HttpMethod.Get, uri) { Version = HttpVersion.Version11 };
```
(although the Version setting could also be removed, since 1.1 is the default)